src/praisonai/praisonai/cli/commands/daemon.py

```
@@ -0,0 +1,163 @@
+
+    import os
+    import signal
+
+    try:
+        os.kill(descriptor.pid, signal.SIGTERM)
+        output.print_success(f"Stopped warm runtime (pid {descriptor.pid}).")
+    except ProcessLookupError:
+        output.print_info("Runtime process already gone; cleaning up lockfile.")
+    except OSError as e:
+        output.print_error(f"Failed to stop runtime: {e}")
+        raise typer.Exit(1)
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)